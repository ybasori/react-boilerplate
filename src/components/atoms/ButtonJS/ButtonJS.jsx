import { sampleData } from "./Button.constant";
import styles from "./ButtonJS.module.scss";
import PropTypes from "prop-types";

const ButtonJS = ({ color, children, onClick }) => {
  return (
    <button
      className={`${styles["container"]} ${styles["color-" + color]}`}
      onClick={onClick}
    >
      {sampleData} {children}
    </button>
  );
};

ButtonJS.propTypes = {
  color: PropTypes.oneOf(["blue", "red"]),
  children: PropTypes.node,
  onClick: PropTypes.func,
};

export default ButtonJS;
