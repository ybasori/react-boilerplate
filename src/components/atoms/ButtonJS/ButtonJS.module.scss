.container{
    padding: 1rem 2rem;
    border-radius: 1rem;
    cursor: pointer;
    font-weight: 600;
    border: none;
}

.color-blue {
    background-color: #3498db;
    color: #fff;
}
.color-red {
    background-color: #e74c3c;
    color: #fff;
}