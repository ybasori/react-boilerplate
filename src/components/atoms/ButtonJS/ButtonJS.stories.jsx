import { fn } from "@storybook/test";
import ButtonJS from "./ButtonJS";

//ðŸ‘‡ This default export determines where your story goes in the story list
const meta = {
  title: "Atoms/ButtonJS",
  component: ButtonJS,
  args: { onClick: fn() },
  argTypes: {
    color: {
      options: ["red", "blue"],
      control: {
        type: "select",
      },
    },
  },
};

export default meta;

export const DefaultButton = {
  args: {
    children: "button",
  },
};
export const BlueButton = {
  args: {
    children: "button",
    color: "blue",
  },
};

export const RedButton = {
  args: {
    children: "button",
    color: "red",
  },
};
