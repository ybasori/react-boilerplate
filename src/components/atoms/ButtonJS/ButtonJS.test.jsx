import { render, screen, fireEvent } from "@testing-library/react";
import ButtonJS from "./ButtonJS";

describe("Atoms/ButtonJS", () => {
  test("renders the button with the correct label", () => {
    render(
      <ButtonJS onClick={() => null} color="blue">
        Click
      </ButtonJS>
    );

    // Assert that the button is in the document and has the correct text
    const buttonElement = screen.getByText("Click");
    expect(buttonElement).toBeInTheDocument();
  });

  test("calls the onClick handler when the button is clicked", () => {
    const handleClick = jest.fn();
    render(
      <ButtonJS onClick={handleClick} color="blue">
        Click
      </ButtonJS>
    );

    const buttonElement = screen.getByText("Click");
    fireEvent.click(buttonElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
