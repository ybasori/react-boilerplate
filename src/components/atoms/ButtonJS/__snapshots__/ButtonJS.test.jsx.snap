// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/ButtonJS Snapshot 1`] = `
<div>
  <button
    class="container color-blue"
  >
    Click
  </button>
</div>
`;
